src/qt/optionsdialog.cpp

```diff
@@ -61,9 +61,15 @@ OptionsDialog::OptionsDialog(QWidget *parent, bool enableWallet) :
     ui->proxyPortTor->setEnabled(false);
     ui->proxyPortTor->setValidator(new QIntValidator(1, 65535, this));
 
-    connect(ui->connectSocks, SIGNAL(toggled(bool)), ui->proxyIp, SLOT(setEnabled(bool)));
-    connect(ui->connectSocks, SIGNAL(toggled(bool)), ui->proxyPort, SLOT(setEnabled(bool)));
-    connect(ui->connectSocks, SIGNAL(toggled(bool)), this, SLOT(updateProxyValidationState()));
+    // Disable and don't connect proxy checkbox interaction if proxy= is set by
+    // a param, to make it clear that removing the param / config is prerequisite to disabling.
+    if (!gArgs.GetArg("-proxy", "").empty()) {
+        ui->connectSocks->setEnabled(false);
+    } else {
+        connect(ui->connectSocks, SIGNAL(toggled(bool)), ui->proxyIp, SLOT(setEnabled(bool)));
```

Comment:
Not a big issue but to me it seems overkill to both disable the control and not connect the signals, isn't only disabling enough?

Reply:
These signal connections re-enable those fields, even if I disable them in the next line. I probably could have fixed with an async call somewhere, but this was easier.